### PR TITLE
HW7: API

### PR DESCRIPTION
**Pull request by skrypetsviacheslav**
[Demo](https://skrypetsviacheslav.github.io/vue-mentoring/)

Some features such as API error handling and loading animations are still in progress. But the main part of the homework is ready, so I decided to create PR and include them as part of the next PR or create a separate PR for them.

### CODE REVIEW

File: src/components/FilmCard.vue [hunk 1]
```diff
@@ -52,6 +52,9 @@ export default {
     onCardClick() {
       console.log("FilmCard#onCardClick");
       this.$emit(EVENTS.FILM_CARD_CLICKED);
+    },
+    setDefaultImage(event) {
+      event.target.src = require(`@/assets/300x450.png`);
```
Comment (Dmitry-White) [inline src/components/FilmCard.vue]: This string should be in "core/constants"

File: src/components/FilmDetailCard.vue [hunk 2]
```diff
@@ -81,6 +81,11 @@ export default {
     description() {
       return this.movie.overview;
     }
+  },
+  methods: {
+    setDefaultImage(event) {
+      event.target.src = require(`@/assets/300x450.png`);
```
Comment (Dmitry-White) [inline src/components/FilmDetailCard.vue]: Same as above.

File: src/core/movieApiService.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+const MovieApiService = {
+  init(options) {
+    movieAxiosInstanse = axios.create({
+      timeout: 5000,
+      ...options,
+      baseURL: "http://react-cdp-api.herokuapp.com"
+    });
+
+    // API doesn't work with single 'filter' query param
```
Comment (Dmitry-White) [inline src/core/movieApiService.js]: If you want to use arrays in query parameters with Axios, take a look at "paramsSerializer" in the config object ("x" in your case)
https://github.com/axios/axios#request-config

**Review by Dmitry-White — CHANGES_REQUESTED** [review]

File: src/core/movieApiService.js (hunk 3 above)
Comment (skrypetsviacheslav) [reply src/core/movieApiService.js]: The API call with a filter parameter works very strange as for me. I even found [sources](https://github.com/avg206/movie-api-server/blob/master/src/api/middleware/filter.js) and at one moment was close to figure out what exactly is wrong :)
So I decided to replace it with simple request with ```searchBy='genres'```and```search='GENRE_NAME'``` params. It looks like everything works in this case.
And I just forgot to remove this interceptor

File: src/components/FilmCard.vue (hunk 1 above)
Comment (skrypetsviacheslav) [reply src/components/FilmCard.vue]: I managed to do it only this way

> const { EVENTS, APP_CONFIG } = require("../config/constants");

> event.target.src = require(`@/${APP_CONFIG.DEFAULT_IMAGE_PATH}`);

In other cases it fails with `module '@/assets/300x450.png' not found` or `APP_CONFIG is not defined`

**Review by Dmitry-White — APPROVED** [review]

**Pull request merged**